### DISCUSSION TITLE
Compile on another Mac, maybe multiple?

### DISCUSSION THREAD

**Discussion by ejelly** [Homebrew/discussions#1591] [Everyday usage]
Hello,

I have slow Macs and fast Macs. Right now I'm upgrading a bunch of homebrew stuff and it takes ages, while my fast Macs are sitting idle.

Is it possible to remotely compile on one of the other Macs? Even better if I could use e.g. distcc to distribute compilation amongst the fast Macs, but if I can just offload it to a single Mac I'd be happy already.

Thanks!

**Comment by carlocab** [comment]
If your faster machines are running the same macOS version as your slower ones, you can use them to build bottles with
```
brew install --build-bottle foo
mkdir /tmp/bottles && cd /tmp/bottles
brew bottle foo
```
This will create bottle tarballs in `/tmp/bottles` that you can copy to your slower ones and install with
```
brew install /path/to/bottle/tarball
```
A better solution is probably to make sure your slower Macs are running at least macOS Mojave, so you should never need to compile anything.

**Comment by ejelly** [comment]
Ah, is there a good way to let one Mac output what's to build, and enter that on the other Mac then? Otherwise I don't know upfront e.g. what dependencies need installing/updating, and entering them by hand is tedious anyway.

As for "never compile anything", my Macs are up to date, and yet right now it's building swig, python, etc... probably because my prefix is not /usr/local/brew.

Can't I just tell homebrew to use distcc as $CC? I have the impression that would solve the problem for everything using a C or C++ compiler, which would be sufficient for me.

**Reply by carlocab** [reply]
Try `brew deps foo` to find all the dependencies of `foo`. `brew` doesn't support using `distcc`. You could maybe hack the code to make it work, but it'll be some work.

**Reply by ejelly** [reply]
Thanks, I will hack a quick shell oneliner together that takes the deps and builds bottles on the remote host.
